keyvi/include/keyvi/dictionary/dictionary.h

```diff
@@ -288,8 +269,8 @@ class Dictionary final {
     MatchIterator current_it = iterators.front();
     iterators.pop();
 
-    auto func = [iterators, current_it]() mutable {
-      while (iterators.size() && (*current_it).IsEmpty()) {
+    auto func = [iterators = std::move(iterators), current_it]() mutable {
+      while (iterators.size() && *current_it == nullptr) {
```

Comment:
` *current_it == nullptr` is another comparison of `match_t` directly to `nullptr`, I believe you could just use the `shared_ptr::operator bool`